update.yml

```diff
@@ -0,0 +1,45 @@
+---
+# Playbook to update all packages on Slurm login and compute nodes
+# The compute nodes are drained before applying updates and optionally
+# rebooting if there is a new kernel.
+
+- hosts: openstack
+  roles:
+    - role: jasmin.cluster-infra
+      cluster_network: "caastest-U-internal"
+      cluster_name: "{{ slurm_name }}"
+      cluster_groups: "{{ slurm_groups }}"
+      cluster_deploy_user: "{{ slurm_deploy_user }}"
+      cluster_net: "{{ slurm_net }}"
+
+- hosts: login
+  gather_facts: no
+  tasks:
+  - import_tasks: drain.yml
+    vars:
+      nodes_to_drain: "{{ groups['compute'] }}"
+
+- hosts:
+    - login
+    - compute
+  become: yes
+  tasks:
+    - import_tasks: openhpc-repos.yml
+    - name: Upgrade all packages
+      yum:
+        name: '*'
+        state: latest
+    - name: Check for reboot hint
+      shell: LAST_KERNEL=$(rpm -q --last kernel | awk 'NR==1 {sub(/kernel-/,""); print $1}'); if [ $LAST_KERNEL != "{{ ansible_kernel }}" ]; then echo 'reboot'; else echo 'no'; fi
+      ignore_errors: true
+      register: reboot_hint
+    - name: Rebooting
+      reboot:
+      when: reboot_hint.stdout.find("reboot") != -1
+
+- hosts: login
+  gather_facts: no
+  tasks:
+  - import_tasks: resume.yml
```

Comment:
As I believe @mkjpryor-stfc suggested elsewhere, we could have a common task file (eg, `slurm-scontrol.yml`) and pass in parameters for `drain` or `resume`.  If that fits and makes sense (it might not).

Reply:
I am not sure if there's much gain in this refactor. Anyway, this can be done in a separate pull request.